genie-agent/src/main/java/com/netflix/genie/agent/execution/services/AgentFileStreamService.java

```diff
@@ -42,6 +44,10 @@
     /**
      * Request the service perform an update outside of the regular schedule to make sure the server has the most
      * up-to-date view of local files.
+     *
+     * @return a scheduled future, in case the caller wants to synchronously wait or verify the operation is successful
+     * or null if the task was not scheduled.
      */
-    void forceServerSync();
+    @Nullable
```

Comment:
Typically in the rest of the codebase we've generally used `Optional` for this kind of thing

Reply:
Switched to Optional, and in the calling code (`RefreshManifestStage`) i'm doing a `get` with timeout, so it waits for a bit but does not fail if it doesn't complete in time.